### ISSUE TITLE
Installation script doesn't use my external drive for data storage

### ISSUE THREAD

**Issue by CookedApps** [nextcloud/vm#964]
Hello, 
I have problems with the installation script not using my external hard drive. The 8TB hard drive is formatted to ext4 and is mounted before I run the script. At the first attempt I selected the drive manually during installation (selected `sdb`), but after the installation finished the Nextcloud data folder was located on my small internal SSD. I reinstalled the entire OS and tried again, this time using the "select automatically" option. Same result. 

Worth to mention is that the script tells me in the beginning that my drive is mounted and that it needs to be unmounted. 

What is the expected behaviour? How do I get the script to use my external drive for Nextcloud data?

## Steps to reproduce
1. Install a fresh Ubuntu 18.04 on an Intel Nuc7
2. Download and run the `nextcloud_install_production.sh` via `sudo bash ...`
3. When the script asks for the desired data location, select "Automatically select" or select the external drive manually

## Expected behaviour
My external drive should be used for data.

## Actual behaviour
My internal SSD is used for storing the data. That can be examined by either taking a look in the Nextcloud admin panel or by searching for the `/nextcloud/data` folder on the drive.

## Server configuration

**Nextcloud server version:** 17.0.0

**Server OS:** Ubuntu 18.04 LTS

**How did you install the VM?** Scripted install from master

## Network
**Do you use DHCP?** Don't know

**Is port 80 and/or 443 open?** Yes

**Comment by enoch85** [comment]
Just skip the format part and add a variable in the installation script that says:

`NCDATA=/your/drive`

You won't get the benefits of ZFS though, if you don't pre-format it in ZFS of course.

**Comment by CookedApps** [comment]
@enoch85 Thanks for the quick response. 
So the part where I select the data drive will only work if my drive is formatted to ZFS? 
What is the recommended procedure regarding the data drive anyway?

**Comment by enoch85** [comment]
In short, it's recommended to run ZFS, and the format script is built upon that the drives aren't mounted already. So to use an external drive it must be either:

1. Unmounted (not possible when using afaik if not ISCSI)
2. Pre-formated and mounted with ZFS or without ZFS with the variable added as in my previous comment - removing the format script code.

> So the part where I select the data drive will only work if my drive is formatted to ZFS?

No, the disk can be anything, as long as it's not mounted.

**Comment by CookedApps** [comment]
Okay so I followed your suggestion number 2. The drive is formatted ext4 and mounted during the installation. I deleted the format script code and added the variable `NCDATA=/media/nuc/RaidExtern/nextcloud`. Additionally I chowned the `nextcloud` folder on `/media/nuc/RaidExtern` to 777, so the script should have alle permissions needed. But nevertheless the script fails as seen on the output on [Pastebin](https://pastebin.com/jmFg0RdT).
